force-app/main/default/classes/ERR_Notifier_TEST.cls

```diff
@@ -0,0 +1,335 @@
+            );
+        }
+    }
+
+    /**********************************************************************************************
+     * @description Confirm that errors are not posted to Chatter if the heap size limit has been reached
+     */
+    @IsTest
+    private static void shouldNotPostErrorsToChatterWithHeapLimit() {
+        if (!Schema.SObjectType.User.isFeedEnabled()) {
+            return;
+        }
+
+        SObject chatterGroup = (SObject) System.Type.forName('CollaborationGroup').newInstance();
+        chatterGroup.put('Name', 'ChatterTestGroup');
+        chatterGroup.put('CollaborationType', 'Private');
+        insert chatterGroup;
+
+        UTIL_CustomSettingsFacade.getSettingsForTests(
+            new Hierarchy_Settings__c(Error_Notifications_On__c = true, Error_Notifications_To__c = chatterGroup.Id)
+        );
+
+        List<Error__c> errors = new List<Error__c>();
+        for (Integer n = 0; n < 100; n++) {
+            errors.add(createError(Datetime.now()));
+        }
+        insert errors;
+
+        ERR_Notifier notifierService = ERR_Notifier.getInstance();
+        Test.startTest();
+        ERR_Notifier.MAX_HEAP_LIMIT = Limits.getHeapSize() * 0.1;
+        notifierService.sendErrorNotifications(null);
+        Test.stopTest();
+
+        for (Error__c err : getErrors()) {
+            System.assertEquals(
+                false,
+                err.Posted_in_Chatter__c,
+                'All errors pending notification should NOT have been posted to chatter because over heap size limit'
+            );
+        }
+    }
+
+    /**********************************************************************************************
+     * @description Confirm that errors are not emailed if the heap size limit has been reached
+     */
+    @IsTest
+    private static void shouldNotEmailErrorsWithHeapLimit() {
+        UTIL_CustomSettingsFacade.getSettingsForTests(
+            new Hierarchy_Settings__c(
+                Error_Notifications_On__c = true,
+                Error_Notifications_To__c = UserInfo.getUserId()
+            )
+        );
+
+        List<Error__c> errors = new List<Error__c>();
+        for (Integer n = 0; n < 100; n++) {
+            errors.add(createError(Datetime.now()));
+        }
+        insert errors;
+
+        ERR_Notifier notifierService = ERR_Notifier.getInstance();
+        //Email has to be sent in sync mode so we can force the heap limit
+        notifierService.sendEmailAsync = false;
+        Test.startTest();
+        ERR_Notifier.MAX_HEAP_LIMIT = Limits.getHeapSize() + 1;
+        notifierService.sendErrorNotifications(null);
+        Test.stopTest();
+
+        for (Error__c err : getErrors()) {
+            System.assertEquals(
+                false,
+                err.Email_Sent__c,
+                'All errors pending notification should NOT have been emailed because over heap size limit'
+            );
+        }
+    }
+
+    // Helpers
```

Comment:
Do we typically add test data methods to `UTIL_UnitTestData_TEST` or is it preferred within the test class?

Reply:
These helper methods are specific for this unit test so I decided to have them here for now